cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -582,6 +582,16 @@ public RunRecordDetail getRun(ProgramReference programRef, String runId) {
     });
   }
 
+  @Override
+  public void markApplicationsLatest(List<ApplicationId> appIds) throws IOException, ConflictException {
+    TransactionRunners.run(transactionRunner, context -> {
+      for (ApplicationId appId: appIds) {
+        ApplicationMeta appMeta = getApplicationMetadata(appId);
+        getAppMetadataStore(context).createApplicationVersion(appId, appMeta);
```

Comment:
should get the store outside the loop so it happens once.

Reply:
should also be calling a new method that just updates the latest flag. Should not be creating a new version or writing any other fields.

Reply:
even though the name of the method `createApplicationVersion` suggests that it creates a new application version, it actually upserts.

Reply:
This has been updated to use the method `markAsLatest` (added in the latest commit). This method just updates the latest field in the appSpec table.